dataprofiler/profilers/profile_builder.py

```diff
@@ -153,18 +155,41 @@ def __add__(self, other):
             options=self.options,
         )
 
+        # Check if indices overlap, if they do, adjust attributes accordingly
+        other_min_id = other.min_id
+        other_max_id = other.max_id
+        other_nti = copy.deepcopy(other.null_types_index)
+        if utils.overlap(self.min_id, self.max_id, other.min_id, other.max_id):
+            warnings.warn(f"Attempted to merge profiles with overlapping "
+                          f"indices. The indices in {other.name} will be "
+                          f"shifted in the merged profile to resolve this.")
+
+            other_min_id = other.min_id + self.max_id + 1
+            other_max_id = other.max_id + self.max_id + 1
+
+            # Increment values (indices) in s by self.max_id + 1
+            def increment(s):
+                return {x + self.max_id + 1 for x in s}
```

Comment:
maybe should be increment set?
Two things:
1. We should probably send in the value in which it is being incremented by
2. should take in input of value by which it is incrementing